prompt2model/dataset_retriever/description_dataset_retriever.py

```diff
@@ -238,41 +232,34 @@ def canonicalize_dataset(self, dataset_name: str) -> datasets.DatasetDict:
                     + f"from {train_columns}\n\n"
                 )
         print(f'Will use the column "{output_column}" as our target.\n')
-        self.print_divider()
+        self._print_divider()
 
         canonicalized_dataset = self.canonicalize_dataset_using_columns(
             dataset, input_columns, output_column
         )
         return canonicalized_dataset
 
-    def retrieve_dataset_dict(
+    def retrieve_top_datasets(
         self,
         prompt_spec: PromptSpec,
-    ) -> datasets.DatasetDict | None:
-        """Select a dataset from a prompt using a dual-encoder retriever.
+    ) -> list[DatasetInfo]:
+        """Retrieve the top datasets for a prompt.
 
         Args:
             prompt_spec: A prompt whose instruction field we use to retrieve datasets.
 
-        Return:
-            A list of relevant datasets dictionaries.
+        Returns:
+            A list of the top datasets for the prompt.
```

Comment:
Hmmm. The definition of top may be ambiguous. Should we summarize how we rank our datasets here?

Reply:
Thanks! I added some more detail.